pkg/plugin/vm_backup_item_action.go

```diff
@@ -81,21 +87,27 @@ func (p *VMBackupItemAction) Execute(item runtime.Unstructured, backup *v1.Backu
 		return nil, nil, fmt.Errorf("VM cannot be safely backed up")
 	}
 
-	skipVolume := func(volume kvcore.Volume) bool {
-		return volumeInDVTemplates(volume, vm)
-	}
-	restore, err := util.RestorePossible(vm.Spec.Template.Spec.Volumes, backup, vm.Namespace, skipVolume, p.log)
-	if err != nil {
-		return nil, nil, errors.WithStack(err)
-	}
-	if !restore {
-		return nil, nil, fmt.Errorf("VM would not be restored correctly")
+	// we can skip all checks that ensure consistency
+	// if we just want to backup for metadata purposes
+	if !isMetadataBackup(backup.ObjectMeta) {
+		skipVolume := func(volume kvcore.Volume) bool {
+			return volumeInDVTemplates(volume, vm)
+		}
+
+		restore, err := util.RestorePossible(vm.Spec.Template.Spec.Volumes, backup, vm.Namespace, skipVolume, p.log)
+		if err != nil {
+			return nil, nil, errors.WithStack(err)
+		}
+		if !restore {
+			return nil, nil, fmt.Errorf("VM would not be restored correctly")
+		}
 	}
 
 	extra, err := kvgraph.NewVirtualMachineBackupGraph(vm)
 	if err != nil {
 		return nil, nil, errors.WithStack(err)
 	}
+	extra = kvgraph.FilterExcludedResources(extra, backup)
```

Comment:
not sure this is what we want.. So far you could have excluded all resources but still get a functioning vm. This is a change in the API.. for example user might want to exclude controller-revisions but will still need CR for the VM to run and he might not even know that..

Reply:
Not sure I understand, so far excluding resources such as pods or PVCs would result in a failed backup. Since now we allow metadata backup we are respecting the excluded resources by giving a backup graph that excludes them. I don't mind changing it but I'm not sure why giving the full backup graph would be more appropriate.

Reply:
Gotcha, I still think it's expected to assume that if a user has excluded certain resources from the backup, these won't pop up in the vm graph. Let's see what @mhenriks thinks.

Reply:
My assumption is that we shouldn't have to do any filtering. If we add something that the user doesn't want, does it get filtered out at a higher level? Have we tested this?

Reply:
Just tested this and the excluded resources are missing from the backup even if they are not filtered by our code. That means the filtering isn't changing any behavior.  Seems good practice to me (consistency with excluded resources) but I'll remove it if you prefer.

Reply:
I think less code on our side the better